.github/workflows/ci.yml

```diff
@@ -101,3 +161,59 @@ jobs:
         uses: actions-rs/cargo@v1.0.3
         with:
           command: test
+
+  tests_macos:
+    name: Run tests macos
+    runs-on: macos-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name:                      Rust Cache
+        uses:                      Swatinem/rust-cache@v1.3.0
+
+      - name: Cargo build
+        uses: actions-rs/cargo@v1.0.3
+        with:
+          command: build
+          args: --workspace
```

Comment:
Do we need to have a build step before the `test` step? (I'll need to look into these rust github actions more!)

Reply:
no you are right, we don't have too but the previous checks only run `cargo check` so we have want make sure nothing weird happens when building the binary but it could be another action for clarity I suppose.

Reply:
My first thought was that `cargo test` builds the binary anyway, but of course it builds it with `cfg(test)` enabled, so I can see why a separate build step is useful to make sure it works "normally"!